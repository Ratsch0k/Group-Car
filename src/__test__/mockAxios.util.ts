jest.mock('axios');
import axios from 'axios';
const mockedAxios = axios as jest.Mocked<typeof axios>;

mockedAxios.head.mockImplementationOnce((path: string) => {
  if (path === '/auth') {
    return Promise.resolve({
      headers: {
        'xsrf-token': 'TEST TOKEN',
      },
    });
  }
});

export default mockedAxios;
