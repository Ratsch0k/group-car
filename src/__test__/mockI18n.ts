jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => undefined),
        languages: ['de', 'en'],
        language: 'en',
      },
    };
  },
}));
