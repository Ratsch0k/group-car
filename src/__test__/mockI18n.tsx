import React from 'react';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => undefined),
        languages: ['de', 'en'],
        language: 'en',
      },
    };
  },
  Trans: (props: any) => {
    return <div>{props.i18nKey}</div>;
  },
}));
