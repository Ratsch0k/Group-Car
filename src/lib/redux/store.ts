import {configureStore, getDefaultMiddleware} from '@reduxjs/toolkit';
import rootReducers from './slices';
import {
  routerMiddleware,
} from 'connected-react-router';
import history from 'lib/redux/history';
import * as Sentry from '@sentry/react';

const sentryReduxEnhancer = Sentry.createReduxEnhancer({
  actionTransformer: (action) => {
    return action;
  },
  stateTransformer: (state) => {
    return state;
  },
});


/**
 * Create redux store.
 */
const store = configureStore({
  reducer: rootReducers(history),
  devTools: true,
  middleware: getDefaultMiddleware().concat(routerMiddleware(history)),
  enhancers: [sentryReduxEnhancer],
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
