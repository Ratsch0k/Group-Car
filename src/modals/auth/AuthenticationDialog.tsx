import {
  Button,
  Dialog,
  DialogContent,
  DialogTitle,
  Divider,
  IconButton,
  makeStyles,
  createStyles,
  Container,
  Typography,
} from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import React, {useState, useEffect, useContext} from 'react';
import {LoginForm} from '../../lib/components/Authentication/Login';
import {useTranslation} from 'react-i18next';
import AuthContext from 'lib/context/auth/authContext';
import CloseIcon from '@material-ui/icons/Close';
import {Route, Switch, useRouteMatch, useLocation} from 'react-router-dom';
import useModalRouter from 'lib/hooks/useModalRouter';
import SignUpBody from 'lib/components/Authentication/SignUp/SignUpBody';

type Theme = import('@material-ui/core').Theme;

const useStyle = makeStyles((theme: Theme) =>
  createStyles({
    content: {
      paddingBottom: theme.spacing(3),
    },
    divider: {
      margin: theme.spacing(2),
    },
    dividerContainer: {
      position: 'relative',
    },
    dividerText: {
      paddingLeft: theme.spacing(0.5),
      paddingRight: theme.spacing(0.5),
      margin: 'auto',
      top: '50%',
      right: '50%',
      position: 'absolute',
      transform: 'translate(50%, -50%)',
      background: theme.palette.background.paper,
    },
    closeButton: {
      float: 'right',
    },
  }),
);

interface AuthenticationDialogProps {
  open: boolean;
  close(): void;
}

const AuthenticationDialog: React.FC<AuthenticationDialogProps> =
(props: AuthenticationDialogProps) => {
  const classes = useStyle();
  const {t} = useTranslation();
  const auth = useContext(AuthContext);
  const {goTo} = useModalRouter();
  const {path, isExact} = useRouteMatch();
  const {pathname} = useLocation();
  const [loading, setLoading] = useState<boolean>(false);

  /**
   * Handles navigation to sign up page
   */
  const handleSignUp = () => {
    goTo(`${path}/sign-up`);
  };

  /**
   * Handles navigation to login page
   */
  const handleLogin = () => {
    goTo(`${path}/login`);
  };

  /**
   * Handles go back action
   */
  const handleBack = () => {
    goTo(undefined);
  };

  /**
   * Navigates to the page from which auth was called.
   */
  const onFinished = () => {
    setLoading(false);
    props.close();
  };

  /**
   * Checks every time the `isLoggedIn` attribute changes
   * if the client is authenticated. If it is go to the origin.
   */
  useEffect(() => {
    if (auth.isLoggedIn) {
      onFinished();
    }
    // eslint-disable-next-line
  }, [auth.isLoggedIn]);

  return (
    <Dialog open={props.open} fullWidth maxWidth='xs'>
      <DialogTitle>
        <IconButton
          id='auth-go-back'
          edge='start'
          disableRipple
          disabled={loading}
          onClick={handleBack}
          style={{visibility: isExact ? 'hidden' : 'visible'}}
        >
          <ArrowBackIcon/>
        </IconButton>
        {
          isExact ?
          t('auth.dialog.title.authenticate') :
          t(`auth.dialog.title.${pathname}`)
        }
        <IconButton
          className={classes.closeButton}
          onClick={onFinished}
          disabled={loading}
          id='auth-close-dialog'
        >
          <CloseIcon />
        </IconButton>
      </DialogTitle>
      <DialogContent className={classes.content}>
        <Switch>
          <Route path={`${path}/login`}>
            <LoginForm
              withSubmit
              onFinished={onFinished}
              setLoading={setLoading}
            />
          </Route>
          <Route path={`${path}/sign-up`}>
            <SignUpBody
              withSubmit
              onFinished={onFinished}
              setLoading={setLoading}
            />
          </Route>
          <Route exact path={path}>
            <Button
              id='auth-to-login'
              fullWidth
              variant='outlined'
              color='primary'
              onClick={handleLogin}
            >
              {t('form.login')}
            </Button>
            <Container className={classes.dividerContainer}>
              <Divider orientation='horizontal' className={classes.divider}/>
              <Typography component='span' className={classes.dividerText}>
                {t('misc.or')}
              </Typography>
            </Container>
            <Button
              id='auth-to-signup'
              fullWidth
              variant='outlined'
              color='primary'
              onClick={handleSignUp}
            >
              {t('form.sign-up')}
            </Button>
          </Route>
        </Switch>
      </DialogContent>
    </Dialog>
  );
};

export default AuthenticationDialog;
