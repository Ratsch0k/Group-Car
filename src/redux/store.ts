import {configureStore} from '@reduxjs/toolkit';
import rootReducer from './slices';

/**
 * Create redux store.
 */
const store = configureStore({
  reducer: rootReducer,
  devTools: true,
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
