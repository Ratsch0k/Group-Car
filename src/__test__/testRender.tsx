import React from 'react';
import {render, RenderResult} from '@testing-library/react';
import {ConnectedRouter} from 'connected-react-router';
import {Provider} from 'react-redux';
import history from '../lib/redux/history';
import {StylesProvider, ThemeProvider} from '@material-ui/styles';
import testTheme from './testTheme';
import configureMockStore, {
  MockStoreCreator,
  MockStoreEnhanced,
} from 'redux-mock-store';
import thunk from 'redux-thunk';
import {RootState} from '../lib/redux/store';
import {
  initialState as authState,
} from '../lib/redux/slices/auth/authSlice';
import {
  initialState as groupState,
} from '../lib/redux/slices/group/groupSlice';
import {
  initialState as invitesState,
} from '../lib/redux/slices/invites/invitesSlice';


export interface TestRenderResult<T> {
  mockStore: MockStoreCreator<T>;
  store: MockStoreEnhanced<unknown>;
}

const defaultState: RootState = {
  router: {
    action: 'POP',
    location: {
      pathname: '/',
      search: '',
      hash: '',
      query: {},
      state: undefined,
    },
  },
  auth: authState,
  group: groupState,
  invites: invitesState,
};

/**
 * Testing generateClassName function to
 * avoid weird fails due to random class names
 * @param rule Rule
 * @param stylesheet Stylesheet
 * @returns The class name
 */
const generateClassName = (rule, stylesheet) => {
  return `${stylesheet.options.classNamePrefix}-${rule.key}`;
};


/**
 * Render function for testing purposes.
 * Provides the necessary base components and
 * contexts.
 * @param children The children to render
 * @returns The result of the render function.
 */
function testRender<T>(state: T, children: React.ReactNode):
RenderResult & TestRenderResult<T> {
  const middleware = [thunk];
  const mockStore = configureMockStore<T>(middleware);
  const initialState = {
    ...defaultState,
    ...state,
  };
  const store = mockStore(initialState);

  const renderResult = render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={testTheme}>
          <StylesProvider generateClassName={generateClassName}>
            {children}
          </StylesProvider>
        </ThemeProvider>
      </ConnectedRouter>
    </Provider>,
  );

  return {
    ...renderResult,
    mockStore,
    store,
  };
}

export default testRender;
