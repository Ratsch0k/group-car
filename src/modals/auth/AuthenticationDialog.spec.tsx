import '../../__test__/mockAxios';
import testRender from '../../__test__/testRender';
import React from 'react';
import {fireEvent} from '@testing-library/react';
import AuthenticationDialog from './AuthenticationDialog';
import {Route} from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { RootState } from '../../lib/redux/store';
import { CALL_HISTORY_METHOD } from 'connected-react-router';
import history from '../../lib/redux/history';
import {User} from "../../typings";

describe('AuthenticationDialog', () => {
  let state: Partial<RootState>;

  beforeEach(() => {
    state = {};
  });

  it('renders without crashing', () => {
    testRender(
      state,
      <AuthenticationDialog open={true} close={() => {}}/>
    );
  });
  
  it('matches snapshot', async () => {
    const {baseElement} = testRender(
      state,
      <AuthenticationDialog open={true} close={() => {}}/>,
    );
  
    expect(baseElement).toMatchSnapshot();
  });
  
  it('sign up button navigates to correct url', () => {
    history.location.pathname = '/auth';
    state = {
      ...state,
      router: {
        location: {
          pathname: '/auth',
          state: undefined,
          search: '',
          query: {},
          hash: '',
        },
        action: 'PUSH'
      },
    };

    const {baseElement, store} = testRender(
      state,
      <Route path='/auth'>
        <AuthenticationDialog open={true} close={() => {}}/>
      </Route>,
    );
  
    expect(baseElement).toMatchSnapshot();
  
    expect(baseElement.querySelector('#auth-to-signup')).not.toBeNull();
    expect(baseElement.querySelector('#auth-to-login')).not.toBeNull();
  
    fireEvent.click(baseElement.querySelector('#auth-to-signup')!);
  
    const expectedAction = {
      type: CALL_HISTORY_METHOD,
      payload: {
        args: [{
          state: {modal: '/auth/sign-up'},
          search: 'modal=/auth/sign-up',
        }],
        method: 'push',
      },
    };

    expect(store.getActions()).toContainEqual(expectedAction);
  });
  
  it('login button navigates to correct url', () => {
    history.location.pathname = '/auth';
    state = {
      ...state,
      router: {
        location: {
          pathname: '/auth',
          state: undefined,
          search: '',
          query: {},
          hash: '',
        },
        action: 'PUSH'
      },
    };

    const {baseElement, store} = testRender(
      state,
      <Route path='/auth'>
        <AuthenticationDialog open={true} close={() => {}}/>
      </Route>,
    );
  
    expect(baseElement).toMatchSnapshot();
  
    expect(baseElement.querySelector('#auth-to-signup')).not.toBeNull();
    expect(baseElement.querySelector('#auth-to-login')).not.toBeNull();
  
    fireEvent.click(baseElement.querySelector('#auth-to-login')!);
  
    const expectedAction = {
      type: CALL_HISTORY_METHOD,
      payload: {
        args: [{
          state: {modal: '/auth/login'},
          search: 'modal=/auth/login',
        }],
        method: 'push',
      },
    };

    expect(store.getActions()).toContainEqual(expectedAction);
  });
  
  it('back button button navigates back from form', () => {
    history.push('/auth');
    history.push('/auth/login');
    state = {
      ...state,
      router: {
        location: {
          pathname: '/auth/login',
          state: undefined,
          search: '',
          query: {},
          hash: '',
        },
        action: 'PUSH'
      },
    };

    const close = jest.fn();

    const {baseElement, store} = testRender(
      state,
      <Route path='/auth'>
        <AuthenticationDialog open={true} close={close}/>
      </Route>
    );
  
    expect(baseElement).toMatchSnapshot();
  
    expect(baseElement.querySelector('#auth-to-signup')).toBeNull();
    expect(baseElement.querySelector('#auth-to-login')).toBeNull();
    expect(baseElement.querySelector('#login-submit')).not.toBeNull();
    expect(baseElement.querySelector('#signup-submit')).toBeNull();
  
    fireEvent.click(baseElement.querySelector('#auth-go-back')!);
  
    const expectedAction = {
      type: CALL_HISTORY_METHOD,
      payload: {
        args: [],
        method: 'goBack'
      },
    };

    expect(store.getActions()).toContainEqual(expectedAction);
    expect(close).not.toHaveBeenCalled();
  });
  
  it('if user logged in call close', () => {
    history.push('/auth');
    history.push('/auth/login');
    state = {
      ...state,
      auth: {
        user: {
          username: 'USER',
          id: 1,
        } as User,
        signUpRequestSent: false,
        loading: false,
      },
      router: {
        location: {
          pathname: '/auth/login',
          state: undefined,
          search: '',
          query: {},
          hash: '',
        },
        action: 'PUSH'
      },
    };

    const close = jest.fn();

    testRender(
      state,
      <Route path='/auth'>
        <AuthenticationDialog open={true} close={close}/>
      </Route>
    );
  
    expect(close).toBeCalledTimes(1);
  });
  
});