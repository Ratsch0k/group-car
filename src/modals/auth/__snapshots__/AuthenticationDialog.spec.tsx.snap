// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationDialog back button button navigates back from form 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
              id="auth-go-back"
              style="visibility: visible;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
            </button>
            auth.dialog.title./auth/login
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton"
              id="auth-close-dialog"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root makeStyles-content"
        >
          <form>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  formik="[object Object]"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-focused MuiInputLabel-focused"
                    data-shrink="true"
                    for="login-username"
                    id="login-username-label"
                  >
                    form.username *
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-focused MuiOutlinedInput-focused MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                  >
                    <input
                      aria-describedby="login-username-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      id="login-username"
                      name="username"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled PrivateNotchedOutline-legendNotched"
                      >
                        <span>
                          form.username *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-focused MuiFormHelperText-marginDense"
                    id="login-username-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  data-testid="textfield"
                  formik="[object Object]"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                    data-shrink="false"
                    for="login-password"
                    id="login-password-label"
                  >
                    form.password *
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                  >
                    <input
                      aria-describedby="login-password-helper-text"
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                      id="login-password"
                      name="password"
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-marginDense"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled"
                      >
                        <span>
                          form.password *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  <p
                    class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-marginDense"
                    id="login-password-helper-text"
                  >
                    <span>
                      ​
                    </span>
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-contained makeStyles-shadowPrimary makeStyles-button MuiButton-containedPrimary MuiButton-fullWidth"
                    id="login-submit"
                    tabindex="0"
                    type="submit"
                  >
                    <span
                      class="MuiButton-label makeStyles-label"
                    >
                      form.login
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AuthenticationDialog login button navigates to correct url 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
              id="auth-go-back"
              style="visibility: hidden;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
            </button>
            auth.dialog.title.authenticate
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton"
              id="auth-close-dialog"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root makeStyles-content"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-login"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.login
            </span>
          </button>
          <div
            class="MuiContainer-root makeStyles-dividerContainer MuiContainer-maxWidthLg"
          >
            <hr
              class="MuiDivider-root makeStyles-divider"
            />
            <span
              class="MuiTypography-root makeStyles-dividerText MuiTypography-body1"
            >
              misc.or
            </span>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-signup"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.sign-up
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AuthenticationDialog matches snapshot 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
              id="auth-go-back"
              style="visibility: hidden;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
            </button>
            auth.dialog.title.authenticate
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton"
              id="auth-close-dialog"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root makeStyles-content"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-login"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.login
            </span>
          </button>
          <div
            class="MuiContainer-root makeStyles-dividerContainer MuiContainer-maxWidthLg"
          >
            <hr
              class="MuiDivider-root makeStyles-divider"
            />
            <span
              class="MuiTypography-root makeStyles-dividerText MuiTypography-body1"
            >
              misc.or
            </span>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-signup"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.sign-up
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`AuthenticationDialog sign up button navigates to correct url 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthXs MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart"
              id="auth-go-back"
              style="visibility: hidden;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
            </button>
            auth.dialog.title.authenticate
            <button
              class="MuiButtonBase-root MuiIconButton-root makeStyles-closeButton"
              id="auth-close-dialog"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
          </h2>
        </div>
        <div
          class="MuiDialogContent-root makeStyles-content"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-login"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.login
            </span>
          </button>
          <div
            class="MuiContainer-root makeStyles-dividerContainer MuiContainer-maxWidthLg"
          >
            <hr
              class="MuiDivider-root makeStyles-divider"
            />
            <span
              class="MuiTypography-root makeStyles-dividerText MuiTypography-body1"
            >
              misc.or
            </span>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
            id="auth-to-signup"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              form.sign-up
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
